coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/parsetree/ParseTreeBuilder.java

```diff
@@ -608,6 +611,96 @@ protected SqlNode visitSelect(ASTNode node, ParseContext ctx) {
     return ctx.selects;
   }
 
+  @Override
+  protected SqlNode visitCreateTable(ASTNode node, ParseContext ctx){
+    CreateTableOptions ctOptions = new CreateTableOptions();
+    for(Node child: node.getChildren()){
+      ASTNode ast = (ASTNode) child;
+      switch (ast.getType()){
+        case HiveParser.TOK_TABNAME:
+          ctOptions.name = (SqlIdentifier) visitTabnameNode(ast, ctx);
+          break;
+        case HiveParser.TOK_IFNOTEXISTS:
+          ctOptions.ifNotExists = true;
+          break;
+        case HiveParser.TOK_TABCOLLIST:
+          ctOptions.columnList = (SqlNodeList) visitColumnList(ast, ctx);
+          break;
+        case HiveParser.TOK_QUERY:
+          ctOptions.query = visitQueryNode(ast, ctx);
+          break;
+        case HiveParser.TOK_TABLESERIALIZER:
+          ctOptions.tableSerializer = visitTableSerializer(ast, ctx);
+          break;
+        case HiveParser.TOK_TABLEFILEFORMAT:
+          ctOptions.tableFileFormat = (SqlNodeList) visitTableFileFormat(ast, ctx);
+          break;
+        case HiveParser.TOK_FILEFORMAT_GENERIC:
+          ctOptions.tableFileFormat = (SqlNodeList) visitFileFormatGeneric(ast, ctx);
+          break;
+        case HiveParser.TOK_TABLEROWFORMAT:
+          ctOptions.tableRowFormat = (SqlCharStringLiteral) visitTableRowFormat(ast, ctx);
+          break;
+        default:
+          break;
+      }
+    }
+    return new SqlCreateTable(ZERO, false, ctOptions.ifNotExists != null ? ctOptions.ifNotExists : false, ctOptions.name,
```

Comment:
Can we introduce a utility class like calcite does here https://github.com/apache/calcite/blob/main/core/src/main/java/org/apache/calcite/sql/ddl/SqlDdlNodes.java to manage all ddl sqlNode creation?

Reply:
@aastha25 Why does DDL SqlNode creation warrant a separate path? Why cannot it be done through generic SqlNode generation?